content/doc/book/pipeline/syntax.adoc

```diff
@@ -697,6 +697,13 @@ pipeline {
 
 The `when` directive allows the Pipeline to determine whether the stage should
 be executed depending on the given condition.
+The `when` directive must contain at least one condition.
+If the `when` directive contains more than one condition,
```

Comment:
@daniel-beck  

This is valid: 
```
when {
    branch 'master'
    environment  name: 'DEPLOY_TO', value: 'production'
}
```
And it is the same as this: 
```
when {
    allOf {
        branch 'master'
        environment  name: 'DEPLOY_TO', value: 'production'
    }
}
```
But I didn't want to say something like "they behave the same as if all the conditions were contained in an `allOf` condition", because we don't explaine the `allOf` condition until the list below.